cai_causal_graph/time_series_causal_graph.py

```diff
@@ -681,7 +673,43 @@ def add_edge(
                 )
         # No else needed as we don't need to check time direction for other edge types.
 
-        edge = super().add_edge(source, destination, edge_type=edge_type, meta=meta, edge=edge, **kwargs)
+    @_reset_ts_graph_attributes
+    def add_edge(
+        self,
+        /,
+        source: Optional[NodeLike] = None,
+        destination: Optional[NodeLike] = None,
+        *,
+        edge_type: EdgeType = EdgeType.DIRECTED_EDGE,
+        meta: Optional[dict] = None,
+        edge: Optional[Edge] = None,
+        **kwargs,
+    ) -> Edge:
+        """
+        Add an edge to the graph. See `cai_causal_graph.causal_graph.CausalGraph.add_edge` for more details.
+
+        In addition to the `cai_causal_graph.causal_graph.CausalGraph.add_edge` method, this method also populates the
+        metadata of the nodes with the variable name and the time lag.
+
+        :param source: The source node of the edge.
+        :param destination: The destination node of the edge.
+        :param edge_type: The type of edge to add.
+        :param meta: The metadata to add to the edge.
+        :param edge: The edge to add.
```

Comment:
Take docstring from base class method as it gives lots of extra info on this parameter.  In fact, I would copy that one in general as it has lots of good info. Prior we just referred the user to that docstring as we still do above but now that all the parameters are copied here too then we should not miss info.